Replace displaced periodic BC test

fdkong opened the pull request:
the inconsistency of the periodic boundary condition when using a displaced system.

The old example has an "Exodiff" issue. The new example is much simpler.

Closes #7946

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 430a7dd2a575d2231f7e62a98480521240fe4c04
+Subproject commit dfcda4f2ed9de7f3d2f4a3221f277879547e5c37
```
**Caution!** This contains a submodule update

fdkong committed:
Replaced the example for testing

the inconsistency of the periodic boundary condition when using a displaced system.

The old example has an "Exodiff" issue. The new example is much simpler.

Closes #7946

moosebuild commented:
Job `App tests:linux-gnu` on 9c116a8 : invalidated by @brianmoose

Yak is fixed now

fdkong commented:
@permcody, any suggestions on this PR? 

@mangerij  are you OK with PR?

fdkong commented:
Thanks, @jwpeterson
